doc/designs/disable-stale-users.md

```diff
@@ -0,0 +1,291 @@
+  DSU:RecordSucessfulAuth-TimeInterval
+* the third one contains a random time period to be added (default: 7 days):
+  DSU:RecordSucessfulAuth-RandomizedTimeInterval
+
+The time period to take into account is 28+7 so 35 days.
+
+As a rule of thumb, if the requirement is to disable users not seen for 90 days
+maximum, updating the attribute every 35 days is good enough.
+At worst, the user would have authenticated the first day and at least once
+during this 35 day period, say day #29. The attribute would therefore contain
+the date from the first day, as day #29 would have been considered too close to
+the existing timestamp to warrant updating it. The user account would then be
+disabled after 61 days of inactivity which fits (loosely) the requirement.
+In short, having a 90-day inactive user disabling requirement would by default
+disable accounts inactive for 55 (90-35) to 90 days. 
+
+Updating this (replicated) timestamp attribute every ~30 days is in fact better
+than mandating password changes every 90 days as the latter would imply
+updating five (replicated) attributes at least once in that 90-day period.
+
+The Disable Stale Users (DSU) tooling, when the above mechanism is active,
+proposes to make sure no non-locked account has a timestamp attribute 90 days
+(by default) in the past.
+
+
+### Passwordless environments
+
+For passwordless environments, S4U2Self is proposed to be added to SSSD to
+account for successful user logon events through ipa-kdb:
+https://pagure.io/SSSD/sssd/issue/4077
+
+The mechanism laid out in "Environments with no mandatory password changes"
+would then be applied as-is.
+
+
+### Notes
+
+Installing the DSU tooling will not change the behavior of existing clusters.
+DSU has to be explicitely configured and enabled to start disabling stale
+accounts.
+
+
+## Background Information
+
+Accounts are locked out by setting the nsAccountLock attribute
+(OID: 2.16.840.1.113730.3.1.610) to True.
+FreeeIPA is not directly involved when a user logs in using an ssh key so it
+cannot make decisions on whether an account is "active" or not. However,
+setting nsaccountlock = TRUE will block all logins including those using ssh
+key authentication.
+
+There is always the possibility that a long-inactive user account will be
+disabled (nsAccountLock'd) by DSU even if that user managed to logon the
+previous few seconds before or during the DSU window due to:
+* the time it takes for the timestamp attribute to be replicated to the DSU
+runner
+* the time it takes for nsAccountLock to be replicated from the DSU runner
+to the replica this user uses.
+
+There is no way to prevent that from happening with the present design, but
+the probability of that happening is low.
+
+
+## Example
+
+### Environments with mandatory, periodic password changes
+
+Given the following requirements and facts:
+* stale accounts must be disabled after 90 days' inactivity
+* grace period is decided to be 8 days
```

Comment:
If the default grace period is 7 days why not use that for the calculation?

Reply:
Fixed, thanks